src/main/java/edu/cornell/kfs/concur/service/impl/ConcurAccountValidationServiceImpl.java

```diff
@@ -2,17 +2,218 @@
+        return validationResult;
+    }
+
+    public ValidationResult checkAccountingString(String chartOfAccountsCode, String accountNumber, String subAccountNumber, String objectCode, String subObjectCode, String projectCode) {
+        ValidationResult requiredFieldsValidationResult = checkRequiredAccountInfo(chartOfAccountsCode, accountNumber, objectCode);
+
+        if (requiredFieldsValidationResult.isNotValid()) {
+            return requiredFieldsValidationResult;
+        } else {
+            ValidationResult accountValidationResult = checkAccount(chartOfAccountsCode, accountNumber);
+            if (accountValidationResult.isNotValid()) {
+                return accountValidationResult;
+            } else {
+                ValidationResult objectCodeValidationResult = checkObjectCode(chartOfAccountsCode, objectCode);
+                ValidationResult subAccountValidationResult = checkSubAccount(chartOfAccountsCode, accountNumber, subAccountNumber);
+                ValidationResult subObjectCodeValidationResult = checkSubObjectCode(chartOfAccountsCode, accountNumber, objectCode, subObjectCode);
+                ValidationResult projectCodeValidationResult = checkProjectCode(projectCode);               
+                ValidationResult validationResult = new ValidationResult(true, new ArrayList<String>());
+
+                if (objectCodeValidationResult.isNotValid()) {
+                    validationResult.setValid(false);
+                    validationResult.addMessages(objectCodeValidationResult.getMessages());
+                }
+
+                if (subAccountValidationResult.isNotValid()) {
+                    validationResult.setValid(false);
+                    validationResult.addMessages(subAccountValidationResult.getMessages());
+                }
+
+                if (subObjectCodeValidationResult.isNotValid()) {
+                    validationResult.setValid(false);
+                    validationResult.addMessages(subObjectCodeValidationResult.getMessages());
+                }
+
+                if (projectCodeValidationResult.isNotValid()) {
+                    validationResult.setValid(false);
+                    validationResult.addMessages(projectCodeValidationResult.getMessages());
+                }
+
+                return validationResult;
+            }
+        }
+    }
+
+    public ValidationResult checkRequiredAccountInfo(String chartOfAccountsCode, String accountNumber, String objectCode) {
+        ValidationResult validationResult = new ValidationResult(true, new ArrayList<String>());
+        if (chartOfAccountsCode == null || chartOfAccountsCode.isEmpty()) {
+            validationResult.setValid(false);
+            validationResult.addMessage(ConcurConstants.AccountingStringValidationErrorMessages.ERROR_CHART_OF_ACCTS_REQUIRED);
+            return validationResult;
+        } else if (accountNumber == null || accountNumber.isEmpty()) {
```

Comment:
Good point, I don't think we need the else if. I mostly used the code from the existing account verification web service and did not think that this would be better without the else if. I will update it.